src/nettests/index.js

```diff
@@ -1,9 +1,106 @@
+import camelCase from 'camelcase'
+
+import percentage from '../cli/output/percentage'
+import wait from '../cli/output/wait'
+
+import {
+  Result,
+  Measurement
+} from '../config/db'
+
 import HTTPHeaderFieldManipulation from './http-header-field-manipulation'
 import HTTPInvalidRequestLine from './http-invalid-request-line'
 import WebConnectivity from './web-connectivity'
 import NDT from './ndt'
 import Base from './base'
 
+export const makeOoni = () => {
+  let dbOperations = [],
+      progress = null,
+      reportId = null,
+      uploaded = false,
+      localReportId = null,
+      isMk = false
+
+  let measurement = Measurement.build({
+    state: 'active'
+  })
+
+  const init = (nt) => {
+    // XXX we use duck typing to see if the argument is a measurement Kit
+    // nettest. Maybe we should change this.
+    if (nt.test && nt.name && nt.setOptions) isMk = true
+
+    measurement.name = camelCase(nt.name)
+    measurement.reportFile = measurement.makeReportFile()
+    dbOperations.push(measurement.save())
+
+    if (isMk) {
+      nt.test.set_options('no_file_report', '0')
+      nt.test.set_output_filepath(measurement.reportFile)
+      nt.on('log', (severity, message) => {
+        // XXX this a workaround due to a bug in MK
+        // I actually also need to know if the report has been created
+        if (message.startsWith('Report ID:') && reportId === null) {
+          reportId = message.split(':')[1].trim()
+          uploaded = true
+        }
+      })
```

Comment:
Is it correct to say that here you only need the logs to find out the `report id`?

Reply:
yes

Reply:
I was thinking about emitting the `report_id` as an `on_event` event rather than having a specific callback for it. Would that be okay for you?